core/src/app.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Self
+
+import ray
+import torch
+from attr import frozen
+from loguru import logger
+
+from core.src.communication.application.godot_app_handler import GodotAppHandler
+from core.src.communication.pipe_handler import PipeHandler
+from core.src.environments.godot_environment import GodotServerEnvironment
+from core.src.settings.app_settings import APP_REGISTRY, AppSettingsSchema
+from core.src.settings.core_settings import CoreSettingsSchema
+from core.src.settings.settings import Environment
+from core.src.setup import configure_logging
+from core.src.utils.registry.shared_registry import SharedRegistry
+from core.src.utils.training_handler import TrainingHandler
+
+
+@frozen
+class App:
+    app_settings: AppSettingsSchema
+    registry: SharedRegistry[type, Any]
+
+    @classmethod
+    def assemble(cls) -> Self:
+        configure_logging()
+        cls.startup_ray()  # Ray has to be started before creating a registry
+        registry: SharedRegistry[type, Any] = SharedRegistry(APP_REGISTRY)
+        app_settings = AppSettingsSchema()
+        registry.put(AppSettingsSchema, app_settings)
+        return cls(app_settings, registry)
+
+    def run(self):
+        if self.app_settings.work_environment.env == Environment.DEVELOPMENT:
+            core_settings = CoreSettingsSchema()
+            self.registry.put(CoreSettingsSchema, core_settings)
+
+            self.train()
+            return
+
+        pipe_name = self.app_settings.work_environment.pipe_name
+        pipe_handler = PipeHandler(pipe_name)
+
+        with GodotAppHandler(pipe_handler) as handler:
+            raw_settings = handler.receive()  # hmm?
+            core_settings = CoreSettingsSchema.model_validate_json(raw_settings)
+            handler.send(bytes(1))
+            self.registry.put(CoreSettingsSchema, core_settings)
+
+            # For now just training?
+            self.train()
+
+    @classmethod
+    def startup_ray(cls):
```

Comment:
Można by to wyrzucić stąd i zrobić klase osobną do ustawień raya (?)

Reply:
i też pipe handler i godot app handler, co by było bradziej czytelne, nie wiem na ile jest to tu potrzebne